src/main/java/com/splunk/hecclient/Indexer.java

```diff
@@ -137,15 +154,63 @@ public boolean send(final EventBatch batch) {
     @Override
     public synchronized String executeHttpRequest(final HttpUriRequest req) {
         CloseableHttpResponse resp;
-        try {
-            resp = httpClient.execute(req, context);
-        } catch (Exception ex) {
-            logBackPressure();
-            log.error("encountered io exception", ex);
-            throw new HecException("encountered exception when post data", ex);
-        }
+        if (!hecConfig.kerberosPrincipal().isEmpty()) {
+            Configuration config = new Configuration() {
```

Comment:
Instead of creating Configuration object for every execute call, can you create it only once and reuse it?